MetaTempUnit/Hardware/library/KelvinVR/src/KelvinVR.cpp

```diff
@@ -187,3 +203,135 @@ void Kelvin::runThermalCycle(bool isCooling, int intensity, int durationSeconds)
 
   Serial.println("Thermal cycle completed.");
 }
+
+void Kelvin::sendStatus()
+{
+  Serial.println("KelvinVR Status:");
+  Serial.println("----------------");
+  Serial.print("Model: ");
+  Serial.println(_modelName);
+  Serial.print("Thermal State: ");
+  if (_isCooling)
+  {
+    Serial.println("Cooling");
+  }
+  else if (_isHeating)
+  {
+    Serial.println("Heating");
+  }
+  else
+  {
+    Serial.println("Idle");
+  }
+  Serial.print("Current Intensity: ");
+  Serial.println(_currentIntensity);
+  Serial.print("LED State: ");
+  Serial.println(_ledState ? "ON" : "OFF");
+  Serial.println("----------------");
+}
+
+void Kelvin::processCommand(const String &command)
+{
+  if (command.startsWith("COOL"))
+  {
+    int intensity = 255;
+    if (command.length() > 5)
+    {
+      intensity = command.substring(5).toInt();
+    }
+    startCooling(intensity);
+    Serial.println("Cooling started");
+  }
+  else if (command.startsWith("HEAT"))
+  {
+    int intensity = 255;
+    if (command.length() > 5)
+    {
+      intensity = command.substring(5).toInt();
+    }
+    startHeating(intensity);
+    Serial.println("Heating started");
+  }
+  else if (command == "STOP")
+  {
+    stopThermalControl();
+    Serial.println("Thermal control stopped");
+  }
+  else if (command.startsWith("CYCLE"))
+  {
+    String params[4];
+    int paramCount = 0;
+    int lastIndex = 0;
+    for (int i = 0; i < command.length(); i++)
+    {
+      if (command.charAt(i) == ' ' || i == command.length() - 1)
+      {
+        params[paramCount++] = command.substring(lastIndex, i == command.length() - 1 ? i + 1 : i);
+        lastIndex = i + 1;
+      }
+    }
+    if (paramCount == 4)
+    {
+      bool isCooling = params[1] == "COOL";
+      int intensity = params[2].toInt();
+      int duration = params[3].toInt();
+      runThermalCycle(isCooling, intensity, duration);
+      Serial.println("Thermal cycle started");
+    }
+  }
+
+  else if (command == "STATUS")
+  {
+    sendStatus();
+  }
+
+  else if (command == "STOPCOOL")
+  {
+    stopCooling();
+    Serial.println("Cooling stopped");
+  }
+  else if (command == "STOPHEAT")
+  {
+    stopHeating();
+    Serial.println("Heating stopped");
+  }
+  else if (command == "LED ON")
+  {
+    ledOn();
+    Serial.println("LED turned on");
+  }
+  else if (command == "LED OFF")
+  {
+    ledOff();
+    Serial.println("LED turned off");
+  }
+  else if (command.startsWith("LED BLINK"))
+  {
+    int delay = 1000;
+    if (command.length() > 10)
+    {
+      delay = command.substring(10).toInt();
+    }
+    blinkFront(delay);
+    Serial.println("LED blinking");
+  }
+  else if (command == "LED FADE BLUE")
```

Comment:
Update README list of commands with updates!

Reply:
Rewrote with switch-case